src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultTagHelperFactsService.cs

```diff
@@ -209,6 +209,19 @@ public override IReadOnlyList<TagHelperDescriptor> GetTagHelpersGivenParent(TagH
             return stringifiedAttributes;
         }
 
+        internal override TagHelperBinding? GetNearestAncestorTagHelperBinding(IEnumerable<SyntaxNode> ancestors)
+        {
+            foreach (var ancestor in ancestors)
+            {
+                if ( ancestor is MarkupTagHelperElementSyntax tagHelperElement)
+                {
+                    return tagHelperElement.TagHelperInfo.BindingResult;
+                }
```

Comment:
Should this explicitly return `null` in the case of a MarkupelementSyntax? That more closely aligns with the behavior in `GetNearestAncestorTagInfo` below, but I'm unsure how far into the nest RestrictChildren is supposed to go.

Reply:
RestrictChildren should only go immedaite parent IIRC. We probably don't need to grab all ancestors